gluten-data/src/main/java/org/apache/gluten/vectorized/ArrowWritableColumnVector.java

```diff
@@ -1255,9 +1266,8 @@ void setNull(int rowId) {
       throw new UnsupportedOperationException();
     }
 
-    void setNotNull(int rowId) {
-      throw new UnsupportedOperationException();
-    }
+    // Arrow not need to setNotNull, set the valus is enough.
+    void setNotNull(int rowId) {}
```

Comment:
Any reason about the change?

Reply:
If we use `MutableProjection`, it will call `MutableColumnarBatchRow`, we can construct `MutableColumnarBatchRow` by `ArrowWritableColumnVector`. https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/MutableColumnarRow.java#L286

Reply:
The class is abstract and the method should be overridable by impl classes.

See https://github.com/apache/incubator-gluten/blob/ed3d333c7c357e95f6736f378c972b714bdfb420/gluten-data/src/main/java/org/apache/gluten/vectorized/ArrowWritableColumnVector.java#L1913 as example.

And abstract class `ArrowVectorWriter` doesn't likely have to be abstract. Could make it an interface as it almost doesn't override anything from the hierarchy.

Reply:
Because it has member `ValueVector`, https://github.com/apache/incubator-gluten/blob/ed3d333c7c357e95f6736f378c972b714bdfb420/gluten-data/src/main/java/org/apache/gluten/vectorized/ArrowWritableColumnVector.java#L1244

Reply:
The class should not be abstract since its subclass will not implement the functions.

Reply:
Can we route `putDecimal` calls to `accessor.setDecimal`? I am not that familiar with the design here but looks to be a manner we should follow

Reply:
writer does not have the `setDecimal` function, I copied most code from `https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java#L429` but is different in putByteArray, because Arrow DecimalVector does not support put byte array range.
Put the all bytes is final function in Spark.
```
public abstract int putByteArray(int rowId, byte[] value, int offset, int count);
  public final int putByteArray(int rowId, byte[] value) {
    return putByteArray(rowId, value, 0, value.length);
  }
```
I will add a test for it.

Reply:
I will override putByteArray but check if it put all bytes, I will refactor it.